scripts/tunnel-creation.mjs

```diff
@@ -0,0 +1,370 @@
+      log.info(`     Device ID: ${device.DeviceID}`);
+      log.info(`     Connection: ${device.Properties.ConnectionType}`);
+      log.info(`     Product ID: ${device.Properties.ProductID}`);
+    });
+
+    let devicesToProcess = devices;
+    if (specificUdid) {
+      devicesToProcess = devices.filter(
+        (device) => device.Properties.SerialNumber === specificUdid,
+      );
+
+      if (devicesToProcess.length === 0) {
+        log.error(`Device with UDID ${specificUdid} not found in connected devices.`);
+        log.error('Available devices:');
+        devices.forEach((device) => {
+          log.error(`  - ${device.Properties.SerialNumber}`);
+        });
+        process.exit(1);
+      }
+    }
+
+    log.info(`\nProcessing ${devicesToProcess.length} device(s)...`);
+
+    /** @type {import('appium-ios-remotexpc').TunnelResult[]} */
+    const results = [];
+
+    for (const device of devicesToProcess) {
+      const result = await this.createTunnelForDevice(device, tlsOptions);
+      results.push(result);
+    }
+
+    log.info('\n=== TUNNEL CREATION SUMMARY ===');
+    const successful = results.filter((r) => r.success);
+    const failed = results.filter((r) => !r.success);
+
+    log.info(`Total devices processed: ${results.length}`);
+    log.info(`Successful tunnels: ${successful.length}`);
+    log.info(`Failed tunnels: ${failed.length}`);
+
+    if (successful.length > 0) {
+      log.info('\n✅ Successful tunnels:');
+      const registry = await this.updateTunnelRegistry(results);
+      await startTunnelRegistryServer(registry, this._tunnelRegistryPort);
+
+      log.info('\n📁 Tunnel registry API:');
+      log.info('   The tunnel registry is now available through the API at:');
+      log.info(`   http://localhost:${this._tunnelRegistryPort}/remotexpc/tunnels`);
+      log.info('\n   Available endpoints:');
+      log.info('   - GET /remotexpc/tunnels - List all tunnels');
+      log.info('   - GET /remotexpc/tunnels/:udid - Get tunnel by UDID');
+      log.info('   - GET /remotexpc/tunnels/metadata - Get registry metadata');
+      if (successful.length > 0) {
+        const firstUdid = successful[0].device.Properties.SerialNumber;
+        log.info(`   curl http://localhost:4723/remotexpc/tunnels/${firstUdid}`);
+      }
+    }
+  }
+}
+
+// Helper function to parse port arguments
+const parsePortArg = (args, flagName, setter) => {
+  const equalsArg = args.find((arg) => arg.startsWith(`${flagName}=`));
+  if (equalsArg) {
+    const port = parseInt(equalsArg.split('=')[1], 10);
+    setter(port);
+    log.info(`Using ${flagName.slice(2)}: ${port}`);
+  } else {
+    const flagIndex = args.indexOf(flagName);
+    if (flagIndex !== -1 && flagIndex + 1 < args.length) {
+      const port = parseInt(args[flagIndex + 1], 10);
+      setter(port);
+      log.info(`Using ${flagName.slice(2)}: ${port}`);
+    }
+  }
+};
+
+// Helper function to parse string arguments
+const parseStringArg = (args, flagName, setter) => {
```

Comment:
is there a reason to use a setter callback and make it more complicated? Can we simply return the parsed value?